lib/Client.js

```diff
@@ -121,8 +121,7 @@ Client.prototype.connect = function connect(options, cb) {
     options = {};
   }
 
-  this._connManager.updateState(options);
-  var connectOptions = this._connManager.getConnectOptions();
+  var connectOptions = this._connManager.updateConnectOptions(options);
```

Comment:
The method `updateConnectOptions` updates all settings so it can be named just `updateOptions`.